src/klein/resource.py

```diff
@@ -75,7 +75,10 @@ def _extractURLparts(request):
     @rtype: L{tuple} of L{unicode}, L{unicode}, L{int}, L{unicode}, L{unicode}
     """
     server_name = request.getRequestHostname()
-    server_port = request.getHost().port
+    if hasattr(request.getHost(), 'port'):
+        server_port = request.getHost().port
+    else:
+        server_port = 0
```

Comment:
Sorry, I didn't notice that it was being returned from the function. Thanks for pointing that out. :-)

A value of zero is fine, but I think it would make sense to add `(True, 0)` and `(False, 0)` to condition below or have some other check to avoid adding `:0` to `server_name` for transports that have no port.